### PR TITLE
Port more tests to munit

### PR DESCRIPTION
**Pull request by cquiroz**
This is another batch of tests ported to munit focused on tests using `Http4sLegacyMatchers`. I hope with another batch I can remove the legacy matcher classes.

### CODE REVIEW

File: client/src/test/scala/org/http4s/client/middleware/FollowRedirectSuite.scala [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+  val defaultClient = Client.fromHttpApp(app)
+  val client = FollowRedirect(3)(defaultClient)
+
+  case class RedirectResponse(
+      method: String,
+      body: String
+  )
+
+  // test("FollowRedirect should ggfollow the proper strategy") {
```
Comment (cquiroz) [inline client/src/test/scala/org/http4s/client/middleware/FollowRedirectSuite.scala:61]: This test was marked as pending and it is a bit hard to port as it uses the `Tables` feature of spec. Instead I just commented it out. Perhaps it should be removed

File: tests/src/test/scala/org/http4s/EntityDecoderSuite.scala [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+    EntityDecoder.decodeText(resp)(implicitly, Charset.`UTF-8`).assertEquals(str)
+  }
+
+  // we want to return a specific kind of error when there is a MessageFailure
+  sealed case class ErrorJson(value: String)
+  implicit val errorJsonEntityEncoder: EntityEncoder[IO, ErrorJson] =
+    EntityEncoder.simple[IO, ErrorJson](`Content-Type`(MediaType.application.json))(json =>
+      Chunk.bytes(json.value.getBytes()))
+
+// TODO: These won't work without an Eq for (Message[IO], Boolean) => DecodeResult[IO, A]
```
Comment (cquiroz) [inline tests/src/test/scala/org/http4s/EntityDecoderSuite.scala:481]: This was commented in the original

File: tests/src/test/scala/org/http4s/EntityDecoderSuite.scala [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+
+  test("Match valid media range") {
+    assert(EntityDecoder.text[IO].matchesMediaType(MediaType.text.plain))
+  }
+
+  test("Match valid media type to a range") {
+    assert(EntityDecoder.text[IO].matchesMediaType(MediaType.text.css))
+  }
+
+  /* TODO: Parameterization
```
Comment (cquiroz) [inline tests/src/test/scala/org/http4s/EntityDecoderSuite.scala:210]: It was commented in the original

File: client/src/test/scala/org/http4s/client/middleware/FollowRedirectSuite.scala (hunk 1 above)
Comment (rossabaker) [reply client/src/test/scala/org/http4s/client/middleware/FollowRedirectSuite.scala:61]: Drop it for now.  #3986 so it can be revived later, if we can.

**Review by rossabaker — APPROVED** [review]
Nice.  This is a thankless task, but I think it's moving us toward a better place.

**Comment by rossabaker** [comment]
Do you want to take care of the merges again?  I already did dotty, because it was trivial.

**Comment by cquiroz** [comment]
Sure I'm do main and cats effect 3 which I guess will still not compile

**Pull request merged**